### PR TITLE
0.91 associated items

### PR DESCRIPTION
**Pull request by korial29**
I've added an Ajax mini-form to add/delete a list of items on ticket.
It's now possible to set multiple items with predefined values (ticket template)
It's also possible to set multiple items with the ticket rules
Migration added to convert templates with the new functionnality

### CODE REVIEW

**Comment by korial29** [comment]
fix #283

**Comment by orthagh** [comment]
There is conflict (sorry for delay), could you fix them ?

Some comments incoming.

File: ajax/dropdownTrackingDeviceType.php [hunk 1]
```diff
@@ -45,37 +45,42 @@
...
     && CommonITILObject::isPossibleToAssignType($_POST["itemtype"])) {
    $table = getTableForItemType($_POST["itemtype"]);
-   $rand  = mt_rand();
-
+   if (isset($_POST["rand"])) {
+      $rand = $_POST["rand"];
+   } else {
+      $rand = mt_rand();
+   }
+
```
Comment (orthagh) [inline ajax/dropdownTrackingDeviceType.php]: Prefers this notation : 

```php
$rand = mt_rand();
if (isset($_POST["rand"])) {
   $rand = $_POST["rand"];
}
```

File: ajax/dropdownTrackingDeviceType.php [hunk 2]
```diff
@@ -45,37 +45,42 @@
...
+   } else {
+      $rand = mt_rand();
+   }
+   
    // Message for post-only
    if (!isset($_POST["admin"]) || ($_POST["admin"] == 0)) {
       echo "<br>".__('Enter the first letters (user, item name, serial or asset number)');
    }
-
+
```
Comment (orthagh) [inline ajax/dropdownTrackingDeviceType.php]: Added trailing spaces

File: ajax/dropdownTrackingDeviceType.php [hunk 3]
```diff
@@ -45,37 +45,42 @@
...
-                              $CFG_GLPI['root_doc']."/ajax/getDropdownFindNum.php",
-                              $p);
+                             $CFG_GLPI['root_doc']."/ajax/getDropdownFindNum.php",
+                             $p);
+   
    // Auto update summary of active or just solved tickets
    $params = array('items_id' => '__VALUE__',
                    'itemtype' => $_POST['itemtype']);
-
+
```
Comment (orthagh) [inline ajax/dropdownTrackingDeviceType.php]: same here

File: inc/ipaddress.class.php [hunk 4]
```diff
@@ -942,15 +942,21 @@ static function getUniqueItemByIPAddress($value, $entity) {
...
+                  $result = array("id"       => $item->getID(),
+                                  "itemtype" => $item->getType());
+                  unset($addressesWithItems);
+                  return $result;
+               }
+            }
+         }
       }
-
+
```
Comment (orthagh) [inline inc/ipaddress.class.php]: ^^

File: inc/networkportinstantiation.class.php [hunk 5]
```diff
@@ -417,15 +416,20 @@ static function getUniqueItemByMac($value, $entity) {
...
+                  $result = array("id"       => $item->getID(),
+                                  "itemtype" => $item->getType());
+                  unset($macs_with_items);
+                  return $result;
+               }
+            }
+         }
       }
-
+
```
Comment (orthagh) [inline inc/networkportinstantiation.class.php]: same here

**Comment by korial29** [comment]
I fixed all conflicts

**Comment by orthagh** [comment]
Some comments (maybe more incoming) : 
- I see some tab spaces in install/update_0901_091.php
- on update process, this error triggers : 
```log
*** PHP Notice(8): Undefined index: use_rich_text
  Backtrace :
  inc/commonitilobject.class.php:2670                
  inc/ticket.class.php:2165                          CommonITILObject->getSearchOptionsMain()
  inc/search.class.php:5284                          Ticket->getSearchOptions()
  install/update_0901_091.php:357                    Search::getOptions()
  install/update.php:791                             update0901to091()
  install/update.php:943                             updateDbUpTo031()
```

**Comment by orthagh** [comment]
The feature seems ok.
After clean of trailing spaces and fix update, we'll can merge

**Comment by orthagh** [comment]
One problem, if you create a ticket from an asset (from ticket tab of a computer for example), an error triggers : 
```log
2016-02-16 09:40:19 [27@adelaunay-ThinkPad-Edge-E320]
  *** PHP Warning(2): Invalid argument supplied for foreach()
  Backtrace :
  inc/item_ticket.class.php:276                      
  inc/ticket.class.php:4159                          Item_Ticket::itemAddForm()
  inc/commonglpi.class.php:464                       Ticket->showForm()
  ajax/common.tabs.php:94                            CommonGLPI::displayStandardTab()
2016-02-16 09:40:19 [27@adelaunay-ThinkPad-Edge-E320]
  *** PHP Warning(2): Invalid argument supplied for foreach()
  Backtrace :
  inc/item_ticket.class.php:285                      
  inc/ticket.class.php:4159                          Item_Ticket::itemAddForm()
  inc/commonglpi.class.php:464                       Ticket->showForm()
  ajax/common.tabs.php:94                            CommonGLPI::displayStandardTab()
```

**Comment by korial29** [comment]
Ok i cleaned the remaining trailing spaces :)

**Comment by orthagh** [comment]
in install/update_0901_091.php, you have always some trailing **tabs** l383-432

**Comment by korial29** [comment]
I fixed the ticket add from assets, and cleaned trailing spaces

**Comment by yllen** [comment]
Undefined index: id in /home/yllen/gitsvn/inc/item_ticket.class.php at line 238

**Pull request merged**